Signal/src/view controllers/MessagesViewController.m

```diff
@@ -202,7 +232,15 @@ -(void) initializeTextView {
     [self.inputToolbar.contentView.textView  setFont:[UIFont ows_regularFontWithSize:17.f]];
     self.inputToolbar.contentView.leftBarButtonItem = _attachButton;
     
-    self.inputToolbar.contentView.rightBarButtonItem = _messageButton;
+    _recordRecognizer = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(recording:)];
+    [_recordButton addGestureRecognizer:_recordRecognizer];
+}
+
+- (NSString *) applicationDocumentsDirectory
```

Comment:
It removes the send button and attaches a long tap gesture recognizer to the record button. Recording is initialized when the user taps and holds on that button.